src/mockups/VaultV2.sol

```diff
@@ -263,8 +269,15 @@ contract VaultV2 {
      * @return success Boolean indicating if there the margin position is successfully decreased.
      * @dev ToDo: Function mainly necessary for integration with untrusted protocols, which is not yet implemnted.
      */
-    function decreaseMarginPosition(address baseCurrency, uint256) public view onlyAuthorized returns (bool success) {
+    function decreaseMarginPosition(address baseCurrency, uint256) public onlyAuthorized returns (bool success) {
         success = baseCurrency == vault.baseCurrency;
+
+        // Update the vault values
+        (address[] memory AllAssetAddresses, uint256[] memory AllAssetIds, uint256[] memory AllAssetAmounts) =
```

Comment:
Can we rename this to ` (address[] memory assetAddresses, uint256[] memory assetIds, uint256[] memory assetAmounts)` to be consistent with other generateAssetData's?